activerecord/test/cases/serialized_attribute_test.rb

```diff
@@ -248,8 +248,8 @@ def test_unexpected_serialized_type
 
     Topic.serialize :content, Array
 
-    topic.reload
     error = assert_raise(ActiveRecord::SerializationTypeMismatch) do
+      topic.reload
```

Comment:
Why did this need to move?

Reply:
I don't know, but it's where the error is happening now.  My gut feeling is that it's okay for the error to be earlier, since it's reasonable at this point to already be aware of the type mismatch.

Reply:
Oh.  I just realized I was forced to make more changes in `reload` after this.  Let me see if I can put it back.